docs/source/deployment/nodes_grouping.md

```diff
@@ -0,0 +1,89 @@
+# Nodes grouping in Kedro: pipelines, tags, and namespaces
+
+Effectively grouping nodes in deployment is crucial for maintainability, debugging, and execution control. This document provides an overview of three key grouping methods: pipelines, tags, and namespaces, along with their strengths, limitations, best uses, and relevant documentation links.
+
+## Grouping by pipelines
+
+If your project contains different pipelines, you can use them as predefined node groupings for deployment. Pipelines can be executed separately in the deployment environment. With the visualisation in Kedro Viz, you can switch to see different pipeline in a more isolated view.
+<br>
+![Switching between different pipelines in Kedro Viz](../meta/images/kedro_viz_switching_pipeline.gif)
+
+If you want to group nodes differently from the existing pipeline structure, you can use tags or namespaces instead of creating a new pipeline. The `kedro run --pipeline` command allows running one pipeline at a time, so multiple pipelines cannot be executed in a single step. While you can switch between pipelines in Kedro Viz, the flowchart view does not support collapsing or expanding them.
+
+**Best used when**
+- You have already separated your logic into different pipelines, and your project is structured to execute them independently in the deployment environment
+
+**Not to use when**
+- You need to run more than one pipeline together.
+- You want to use the expand and collapse functionality in Kedro Viz
+
+**How to use**
+
+  ```bash
+  kedro run --pipeline=<your_pipeline_name>
+  ```
+More information: [Run a pipeline by name](https://docs.kedro.org/en/stable/nodes_and_pipelines/run_a_pipeline.html#run-a-pipeline-by-name)
+
+---
+
+## Grouping by tags
+
+You can tag individual nodes or the entire pipeline, allowing flexible execution of specific sections without modifying the pipeline structure.Kedro-Viz provides a clear visualisation of tagged nodes, making it easier to understand.
+<br>
+![Filters Panel in Kedro Viz](../meta/images/kedro_viz_filters_tags.png)
+
+Please note that nodes with the same tag can exist in different pipelines, making debugging and maintaining the codebase more challenging.
+
+**Best used when**
+- You need to run specific nodes that don’t belong to the same pipeline.
+- You want to rerun a subset of nodes in a large pipeline.
+
+**Not to use when**
+- The tagged nodes have strong dependencies, which might cause execution failures.
+- Tags are not hierarchical, so tracking groups of nodes can become difficult.
+- Tags do not enforce structure like pipelines or namespaces.
```

Comment:
I think this last point isn't so much a `Not to use when` but more a note on tags. Maybe it makes sense to mention this before the best used when/not to use when sections.

Reply:
DONE 👍